### PR TITLE
Add the close function when load plugins

### PR DESCRIPTION
**Pull request by xiangge**
Add the close function when load plugins
JIRA: PDC-2235

### CODE REVIEW

File: pdc_client/runner.py [hunk 1]
```diff
@@ -110,14 +110,19 @@ def load_plugins(self):
...
+                    file, pathname, description = imp.find_module(name[:-3], [dir])
+                    plugin = imp.load_module(name[:-3], file, pathname, description)
+                    self.logger.debug('Loaded plugin {0}'.format(name[:-3]))
+                    self.raw_plugins.append(plugin)
+                    if hasattr(plugin, 'PLUGIN_CLASSES'):
+                        for p in plugin.PLUGIN_CLASSES:
+                            self.logger.debug('Instantiating {0}'.format(p.__name__))
+                            self.plugins.append(p(self))
+                except:
+                    print('Failed to init plugin %s.' % name[:-3])
```
Comment (simozhan) [inline pdc_client/runner.py]: 1. Maybe 'Failed to load plugin'
2. Is it ok to make consistent style with other log, like 'Loaded plugin {0}'.format...

**Review by simozhan — CHANGES_REQUESTED** [review]

File: pdc_client/runner.py (hunk 1 above)
Comment (xiangge) [reply pdc_client/runner.py]: Changed, please review

**Review by simozhan — APPROVED** [review]

File: pdc_client/runner.py [hunk 2]
```diff
@@ -110,14 +110,19 @@ def load_plugins(self):
...
+                    file, pathname, description = imp.find_module(name[:-3], [dir])
+                    plugin = imp.load_module(name[:-3], file, pathname, description)
+                    self.logger.debug('Loaded plugin {0}'.format(name[:-3]))
+                    self.raw_plugins.append(plugin)
+                    if hasattr(plugin, 'PLUGIN_CLASSES'):
+                        for p in plugin.PLUGIN_CLASSES:
+                            self.logger.debug('Instantiating {0}'.format(p.__name__))
+                            self.plugins.append(p(self))
+                except:
+                    print('Failed to load plugin {0}'.format(name[:-3]))
```
Comment (hluk) [inline pdc_client/runner.py]: Omit `print()` for status and error messages. And try to show more information.
```python
except Exception as e:
    self.logger.error('Failed to load plugin "{0}": {1}'.format(name[:-3], e))
```

File: pdc_client/runner.py [hunk 3]
```diff
@@ -110,14 +110,19 @@ def load_plugins(self):
...
+                finally:
+                    file.close()
```
Comment (hluk) [inline pdc_client/runner.py]: Use `if file:` here.

File: pdc_client/runner.py [hunk 4]
```diff
@@ -110,14 +110,19 @@ def load_plugins(self):
...
-                file, pathname, description = imp.find_module(name[:-3], [dir])
-                plugin = imp.load_module(name[:-3], file, pathname, description)
-                self.logger.debug('Loaded plugin {0}'.format(name[:-3]))
-                self.raw_plugins.append(plugin)
-                if hasattr(plugin, 'PLUGIN_CLASSES'):
-                    for p in plugin.PLUGIN_CLASSES:
-                        self.logger.debug('Instantiating {0}'.format(p.__name__))
-                        self.plugins.append(p(self))
+                try:
+                    file, pathname, description = imp.find_module(name[:-3], [dir])
```
Comment (hluk) [inline pdc_client/runner.py]:
```python
module_name = name[:-3]
```

**Review by hluk — CHANGES_REQUESTED** [review]

File: pdc_client/runner.py (hunk 4 above)
Comment (xiangge) [reply pdc_client/runner.py]: Changed.

File: pdc_client/runner.py (hunk 2 above)
Comment (xiangge) [reply pdc_client/runner.py]: I think use print here more makes sense, we have no need to show a "Error:pdc: ......" when ct meets error in command lines, but the error info should be included.
Changed.

File: pdc_client/runner.py (hunk 3 above)
Comment (xiangge) [reply pdc_client/runner.py]: changed, but do you think it's possible to get a None value of file here?

File: pdc_client/runner.py (hunk 2 above)
Comment (hluk) [reply pdc_client/runner.py]: With `print()` you're printing on stdout by default, I wouldn't expect any errors there. Let logger handle the details. Also it makes messages to follow some format and I can grep errors easily.

Comment (xiangge) [reply pdc_client/runner.py]: # Also it makes messages to follow some format and I can grep errors easily.
Make sense, change

Comment (xiangge) [reply pdc_client/runner.py]: Changed, please help to review, will commit another patch to change other print

**Review by hluk — APPROVED** [review]

**Pull request merged**